src/storage/mod.rs

```diff
@@ -4202,4 +4250,78 @@ mod tests {
             .unwrap();
         rx.recv().unwrap();
     }
+
+    #[test]
+    fn test_txn_heart_beat() {
+        let storage = TestStorageBuilder::new().build().unwrap();
+        let (tx, rx) = channel();
+
+        // No lock.
+        storage
+            .async_txn_heart_beat(
+                Context::default(),
+                Key::from_raw(b"k"),
+                10,
+                100,
+                expect_fail_callback(tx.clone(), 0, |e| match e {
+                    Error::Txn(txn::Error::Mvcc(mvcc::Error::TxnLockNotFound { .. })) => (),
+                    e => panic!("unexpected error chain: {:?}", e),
+                }),
+            )
+            .unwrap();
+        rx.recv().unwrap();
+
+        let mut options = Options::default();
+        options.lock_ttl = 100;
+        storage
+            .async_prewrite(
+                Context::default(),
+                vec![Mutation::Put((Key::from_raw(b"k"), b"v".to_vec()))],
+                b"k".to_vec(),
+                10,
+                options,
+                expect_ok_callback(tx.clone(), 0),
+            )
+            .unwrap();
+        rx.recv().unwrap();
+
+        // Less TTL, not updated.
+        storage
+            .async_txn_heart_beat(
+                Context::default(),
+                Key::from_raw(b"k"),
+                10,
+                90,
```

Comment:
how about define 
`
   advice_ttl=90;
   expect_ttl=100
`
to make the test case more easily to understand?

Reply:
I think it may be still difficult to read since you need to find the variable's definition or assignment to know what value it is. I think I'll add comments near these arguments.